module/zfs/qat_compress.c

```diff
@@ -379,6 +413,7 @@ qat_compress(qat_compress_dir_t dir, char *src, int src_len,
 
 		compressed_sz = dc_results.produced;
 		if (compressed_sz + hdr_sz + ZLIB_FOOT_SZ > dst_len) {
```

Comment:
this check tries to make sure there is enough space in dest buffer for the compressed data and gzip head + foot,  "add_len" should be added. Actually I'm now thinking maybe we can remove this check safely, as adding the addition buffer, the dest size is large enough to contain all output in theory, it is 1.75x of source buffer now!

Reply:
i'd rather keep this in in case this code gets used in other contexts.

Reply:
OK!

Reply:
Please add add_len

Reply:
I didn't want to do that because in my comment I say that the `add_buf` is a scratch buffer that can be discarded after the function is done, and if we leak into the add_buf we know we're over the useful limit anyway.